src/GraphQLParser.ApiTests/GraphQLParser.approved.txt

```diff
@@ -661,6 +661,7 @@ namespace GraphQLParser
     public static class Parser
     {
         public static GraphQLParser.AST.GraphQLDocument Parse(GraphQLParser.ROM source, GraphQLParser.ParserOptions options = default) { }
+        public static GraphQLParser.AST.GraphQLValue ParseValue(GraphQLParser.ROM source, GraphQLParser.ParserOptions options = default) { }
```

Comment:
Just a note - it seems that we could make more methods public to allow parse concrete parts of document.

Reply:
Yeah I was thinking like "couldn't we just have Parse" return whatever was parsed?  Then cast it to whatever type was desired or something?  But the problem is that the parser wouldn't know the initial state -- if, for example, it was parsing a list of arguments or a list of directives.

We could perhaps add `Parse<T>` where T is a node type, and then a `switch typeof(T)` logic to bump into the correct code.

Reply:
Then you could write `Parse<GraphQLValue>` if you want that, or `Parse<List<GraphQLArgument>>` if you are parsing that, etc, with very little API changes and somewhat intuitive behavior.

Reply:
Not really a fan of a million different public `Parse...` methods.  Hopefully they are all exposed as protected methods or something so anyone CAN use them if needed.

Reply:
This is a subject to discuss for v9.

Reply:
#296